src/app/app.component.ts

```diff
@@ -1,34 +1,59 @@
-import { Component } from "@angular/core";
-import { mockedAuthorsList, mockedCoursesList } from "../assets/mocks/mocks";
-import { Course, CourseInfo } from "./shared/types/course.model";
-import { Author } from "./shared/types/author.model";
-import { MappingService } from "./services/mapping.service";
-import { ButtonTypes } from "./shared/types/button.type";
-import { FaIconLibrary } from "@fortawesome/angular-fontawesome";
-import { fas } from "@fortawesome/free-solid-svg-icons";
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ButtonTypes } from './shared/types/button.type';
+import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
+import { fas } from '@fortawesome/free-solid-svg-icons';
+import { AuthService } from './auth/services/auth.service';
+import { Router } from '@angular/router';
+import { UserStoreService } from './user/services/user-store.service';
+import { SessionStorageService } from './auth/services/session-storage.service';
+import { CoursesStoreService } from './services/courses-store.service';
+import { Subscription } from 'rxjs';
 
 @Component({
-  selector: "app-root",
-  templateUrl: "./app.component.html",
-  styleUrls: ["./app.component.scss"],
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss'],
 })
-export class AppComponent {
+export class AppComponent implements OnInit, OnDestroy {
   ButtonTypes = ButtonTypes;
-
-  mockedCourses: Course[] = mockedCoursesList;
-  mockedAuthors: Author[] = mockedAuthorsList;
-  courses: CourseInfo[] = this.mapping.createCoursesWithAuthorNames(
-    this.mockedCourses,
-    this.mockedAuthors,
-  );
+  private subscription: Subscription | undefined;
 
   constructor(
-    private mapping: MappingService,
     private library: FaIconLibrary,
+    private authService: AuthService,
+    private router: Router,
+    private userService: UserStoreService,
+    private token: SessionStorageService,
+    private coursesStore: CoursesStoreService
   ) {
     this.library.addIconPacks(fas);
   }
 
-  title = "courses-app";
-  user = "Admin";
+  ngOnInit() {
+    this.userService.getUser();
+    this.coursesStore.getAll();
+    this.coursesStore.getAllAuthors();
+    if (this.token.getToken()) this.authService.isAuthorised = true;
+  }
+
+  onLogout = () => {
```

Comment:
Stick with the regular function signature instead of arrow function.

Reply:
done. somehow it became an arrow func : )